exporter/exporterhelper/common.go

```diff
@@ -21,7 +21,7 @@ import (
 // requestSender is an abstraction of a sender for a request independent of the type of the data (traces, metrics, logs).
 type requestSender interface {
 	component.Component
-	send(context.Context, Request) error
+	send(context.Context, ...Request) error
```

Comment:
Can you please elaborate on why this change is required? I cannot find a reason for it to support concurrency limit.

Reply:
Thanks for the review! The main issue comes from returning multiple errors back to a single batch producer. e.g. let's say one request enters the batch_sender and is split into 50 batches. We can't return from sendMergeSplitBatch until we collect a response from all 50 batch exports. In the current implementation we [sequentially](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/batch_sender.go#L190) export these batches waiting for the previous batch to return -- ideally these exports should be concurrent subject to a concurrency limit. To make this concurrent we want to avoid creating too many goroutines that exceeds the concurrency limit. This is reason we need to pass the full slice of requests to the concurrency_sender so it can limit the number of goroutines used for export